pynicotine/plugins/leech_detector/__init__.py

```diff
@@ -71,72 +76,83 @@ def loaded_notification(self):
         if self.settings["num_folders"] < min_num_folders:
             self.settings["num_folders"] = min_num_folders
 
-        if self.settings["message"]:
-            self.str_action = "message leecher"
-        else:
-            self.str_action = "log leecher"
+        self.messaged_users = set(self.settings["messaged_users"])
 
         self.log(
-            "Ready to %ss, require users have a minimum of %d files in %d shared public folders.",
-            (self.str_action, self.settings["num_files"], self.settings["num_folders"])
+            "Require users have a minimum of %d files in %d shared public folders.",
+            (self.settings["num_files"], self.settings["num_folders"])
         )
 
     def upload_queued_notification(self, user, virtual_path, real_path):
 
-        if user in self.probed:
+        if user in self.messaged_users:
+            # Already messaged user previously
```

Comment:
What about if the user has since gained more files, then should we remove them from the list?

Reply:
I suppose we could do that.